middleware/file/source/manager/admin/cli.cpp

```diff
@@ -0,0 +1,179 @@
+                        common::terminal::format::column( "pid", []( const auto& r){ return r.pid;}, common::terminal::color::white),
+                        common::terminal::format::column( "gtrid", []( const auto& r){ return r.gtrid;}),
+                        common::terminal::format::column( "stage", []( const auto& r){ return description( r.stage);}),
+                        common::terminal::format::column( "path", []( const auto& r){ return r.path;}, common::terminal::color::yellow));
+                  }
+               } // format
+
+            } // detail
+
+            namespace list
+            {
+               namespace reservations
+               {
+                  auto option()
+                  {
+                     auto invoke = []()
+                     {
+                        auto state = detail::call::state();
+
+                        auto formatter = detail::format::requests( state);
+
+                        formatter.print( std::cout, state.requests);
+                     };
+
+                     return argument::Option{
+                        std::move( invoke),
+                        { "-r", "--list-reservations"},
+                        "list information of files currently reserved"
+                     };
+                  }
+               } // reservations
+
+            } // list
+
+            namespace assets
+            {
+               
+               namespace recovery
+               {
+                  namespace detail
+                  {
+                     auto invoke( const manager::admin::model::recovery::Directive directive)
+                     {
+                        return [directive]( common::transaction::global::ID gtrid, std::vector< common::transaction::global::ID> gtrids)
+                        {
+                           gtrids.insert( std::begin( gtrids), std::move( gtrid));
+                           const auto recovered = local::detail::call::recover( std::move( gtrids), directive); 
+
+                           for( const auto& gtrid : recovered)
+                           {
+                              common::log::line( std::cout, gtrid);
+                           }
+                        };
+                     }
+
+                     auto complete()
+                     {
+                        return []( bool help, auto values) -> std::vector< std::string>
+                        {
+                           if( help)
+                              return { "<gtrid>"};
+
+                           std::vector< std::string> result;
+
+                           std::ranges::transform( 
+                              local::detail::call::state().requests,
+                              std::back_inserter( result),
+                              []( const auto& request) { return common::string::compose( request.gtrid); });
+
```

Comment:
Why not use `algorithm::transform(...)` that returns an std::vector with the return type of the functor? Hence, you can do a "one-liner".
The most common use-case for transform in `casual` is exactly this -> we have our own algorithm for it.